---
layout: chapter
title: Models of discrete character evolution
---
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
  <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
  <meta http-equiv="Content-Style-Type" content="text/css" />
  <meta name="generator" content="pandoc" />
  <title></title>
  <style type="text/css">code{white-space: pre;}</style>
</head>
<body>
<div id="TOC">
<ul>
<li><a href="#chapter-7-models-of-discrete-character-evolution">Chapter 7: Models of discrete character evolution</a><ul>
<li><a href="#biological-motivation-limblessness-as-a-discrete-trait">Biological motivation: Limblessness as a discrete trait</a></li>
<li><a href="#modeling-the-evolution-of-discrete-states">Modeling the evolution of discrete states</a></li>
<li><a href="#the-mk-model">The Mk Model</a></li>
<li><a href="#the-extended-mk-model">The Extended Mk Model</a></li>
<li><a href="#simulating-the-mk-model-on-a-tree">Simulating the Mk model on a tree</a></li>
<li><a href="#chapter-summary">Chapter summary</a></li>
<li><a href="#references">References</a></li>
</ul></li>
</ul>
</div>
<p><a href="{{site.baseurl}}/pdf/chapter7_introdiscrete.pdf">pdf version</a></p>
<h1 id="chapter-7-models-of-discrete-character-evolution">Chapter 7: Models of discrete character evolution</h1>
<h2 id="biological-motivation-limblessness-as-a-discrete-trait">Biological motivation: Limblessness as a discrete trait</h2>
<p>Squamates, the clade that includes all living species of lizards, are well known for their diversity. From the gigantic Komodo dragon of Indonesia (Figure 7.1A, <em>Varanus komodoensis</em>) to tiny leaf chameleons of Madagascar (Figure 7.1B, <em>Brookesia</em>), squamates span an impressive range of form and ecological niche use <span class="citation">(Vitt et al. <a href="#ref-Vitt2003-bl">2003</a>; Pianka et al. <a href="#ref-Pianka2017-aw">2017</a>)</span>. Even the snakes (Figure 7.1C and D), extraordinarily diverse in their own right (~3,500 species), are actually a clade that is nested within squamates <span class="citation">(Streicher and Wiens <a href="#ref-Streicher2017-wb">2017</a>)</span>. The squamate lineage that is ancestral to snakes became limbless about 170 million years ago (see <a href="www.timetree.org">Timetree of Life</a>) – and also underwent a suite of changes to their head shape, digestive tract, and other traits associated with their limbless lifestyle. In other words, snakes are lizards – highly modified lizards, but lizards nonetheless. And snakes are not the only limbless lineage of squamates. In fact, lineages within squamates have lost their limbs over and over again through their history (e.g. Figure 7.1E and F), with some estimates that squamates have lost their limbs at least 26 times in the past 240 million years <span class="citation">(Brandley et al. <a href="#ref-Brandley2008-wr">2008</a>)</span>.</p>
<div class="figure">
<img src="{{site.baseurl}}/images/figure7-1.png" alt="Figure 7.1. Squamates, legged and legless. A. Komodo dragon, B. Brookesia chameleon, C. and D. snakes, E. and F. legless lizards." />
<p class="caption">Figure 7.1. Squamates, legged and legless. A. Komodo dragon, B. Brookesia chameleon, C. and D. snakes, E. and F. legless lizards.</p>
</div>
<p>Limblessness is an example of a discrete trait – a trait that can occupy one of a set of distinct character states. Analyzing the evolution of discrete traits requires a different modeling approach than what we used for continuous traits. In this chapter, I will discuss the Mk model <span class="citation">(Lewis <a href="#ref-Lewis2001-bu">2001</a>)</span>, which is a general approach to modeling the evolution of discrete traits on trees. Fitting this model to comparative data will help us understand the evolution of traits like limblessness where species can be placed into one of a number of discrete character states.</p>
<p>Key Questions</p>
<ul>
<li><p>How can we model the evolution of discrete characters that have a set number of fixed states?</p></li>
<li><p>How can we change the parameters of the Mk model to construct more elaborate models of discrete character evolution?</p></li>
<li><p>How do we simulate the evolution of a discrete character?</p></li>
</ul>
<h2 id="modeling-the-evolution-of-discrete-states">Modeling the evolution of discrete states</h2>
<p>So far, we have only dealt with continuously varying characters. However, many characters of interest to biologists can best be described by characters with a set number of fixed states. For limblessness in squamates, introduced above (snakes and lizards), each species is either legless (state 0) or not <span class="citation">(state 1; actually, there are some species that might be considered “intermediate” Brandley et al. <a href="#ref-Brandley2008-wr">2008</a>, but we will ignore those here)</span>. We might have particular questions about the evolution of limblessness in squamates. For example, how many times this character has changed in the evolutionary history of squamates? How often does limblessness evolve? Do limbs ever re-evolve? Is the evolution of limblessness related to some other aspect of the lives of these reptiles?</p>
<p>We will consider discrete characters where each species might exhibit one of <span class="math inline"><em>k</em></span> states. (In the limbless example above, <span class="math inline"><em>k</em> = 2</span>). For characters with more than two states, there is a key distinction between ordered and unordered characters. Ordered characters can be placed in an order so that transitions only occur between adjacent states. For example, I might include “intermediate” species that are somewhere in between limbed and limbless – for example, the “mermaid skinks” (Sirenoscincus) from Madagascar, so called because they lack hind limbs <span class="citation">(Figure 7.2, Moch and Senter <a href="#ref-Moch2011-fp">2011</a>)</span>. An ordered model might only allow transitions between limbless and intermediate, and intermediate and limbed; it would be impossible under such a model to go directly from limbed to limbless without first becoming intermediate. For unordered characters, any state can change into any other state. In this chapter, I will focus mainly on unordered characters; we will return to ordered characters later in the book.</p>
<div class="figure">
<img src="{{site.baseurl}}/images/figure7-2.png" alt="Figure 7.2. Mermaid skink" />
<p class="caption">Figure 7.2. Mermaid skink</p>
</div>
<p>Most work on the evolution of discrete characters on phylogenetic trees has focused on the evolution of gene or protein sequences. Gene sequences are made up of four character states (A, C, T, and G for DNA). Models of sequence evolution allow transitions among all of these states at certain rates, and may allow transition rates to vary across sites, among clades, or through time. There are a huge number of named models that have been applied to this problem <span class="citation">(e.g. Jukes-Cantor, JC; General Time-Reversible, GTR; and many more, Yang <a href="#ref-Yang2006-md">2006</a>)</span>, and a battery of statistical approaches are available to fit these models to data <span class="citation">(e.g. Posada <a href="#ref-Posada2008-qw">2008</a>)</span>.</p>
<p>Any discrete character can be modeled in a similar way as gene sequences. When considering phenotypic characters, we should keep in mind two main differences from the analysis of DNA sequences. First, arbitrary discrete characters may have any number of states (beyond the four associated with DNA sequence data). Second, characters are typically analyzed independently rather than combining long sets of characters and assuming that they share the same model of change.</p>
<h2 id="the-mk-model">The Mk Model</h2>
<p>The most basic model for discrete character evolution is called the Mk model. First developed for trait data by Pagel <span class="citation">(<a href="#ref-Pagel1994-ui">1994</a>; although the name Mk comes from Lewis <a href="#ref-Lewis2001-bu">2001</a>)</span>. The Mk model is a direct analogue of the Jukes-Cantor (JC) model for sequence evolution. The model applies to a discrete character having <span class="math inline"><em>k</em></span> unordered states. Such a character might have <span class="math inline"><em>k</em> = 2</span>, <span class="math inline"><em>k</em> = 3</span>, or even more states. Evolution involves changing between these <span class="math inline"><em>k</em></span> states (Figure 7.3).</p>
<div class="figure">
<img src="{{site.baseurl}}/images/figure7-3.png" alt="Figure 7.3. Examples of discrete characters with (A) k = 2, (B) k = 3, and (C) k = 4 states." />
<p class="caption">Figure 7.3. Examples of discrete characters with (A) <span class="math inline"><em>k</em> = 2</span>, (B) <span class="math inline"><em>k</em> = 3</span>, and (C) <span class="math inline"><em>k</em> = 4</span> states.</p>
</div>
<p>The basic version of the Mk model assumes that transitions among these states follow a Markov process. This means that the probability of changing from one state to another depends only on the current state, and not on what has come before. For example, it makes no difference if a lineage has just evolved the trait of “feathers,” or whether they have had feathers for millions of years – the probability of evolving a different character state is the same in both cases. The basic Mk model also assumes that every state is equally likely to change to any other states.</p>
<p>For the basic Mk model, we can denote the instantaneous rate of change between states using the parameter <span class="math inline"><em>q</em></span>. In general, <span class="math inline"><em>q</em><sub><em>i</em><em>j</em></sub></span> is called the instantaneous rate between character states <span class="math inline"><em>i</em></span> and <span class="math inline"><em>j</em></span>. It is defined as the limit of the rate measured over very short time intervals. Imagine that you calculate a rate of character change by counting the number of changes of state of a character over some time interval, <span class="math inline"><em>t</em></span>. You can calculate the rate of change as <span class="math inline"><em>n</em><sub><em>c</em><em>h</em><em>a</em><em>n</em><em>g</em><em>e</em><em>s</em></sub>/<em>t</em></span>. The instantaneous rate is the value that this rate approaches as <span class="math inline"><em>t</em></span> gets smaller and smaller so that the time interval is nearly zero. Again, for the basic Mk model, instantaneous rates between all pairs of characters are equal; that is, <span class="math inline"><em>q</em><sub><em>i</em><em>j</em></sub> = <em>q</em><sub><em>m</em><em>n</em></sub></span> for all <span class="math inline"><em>i</em> ≠ <em>j</em></span> and <span class="math inline"><em>m</em> ≠ <em>n</em></span>.</p>
<p>We can summarize general Markov models for discrete characters using a transition rate matrix <span class="citation">(Lewis <a href="#ref-Lewis2001-bu">2001</a>)</span>:</p>
(eq. 7.1)
<div>
<p><br /><span class="math display">$$
\mathbf{Q} =
\begin{bmatrix}
-d_1 &amp; q_{12} &amp; \dots &amp; q_{1k} \\
q_{21} &amp; -d_2 &amp; \dots &amp; q_{2k} \\
\vdots &amp; \vdots &amp; \ddots &amp; \vdots\\
q_{k1} &amp; q{k2} &amp; \dots &amp; -d_k \\
\end{bmatrix}
$$</span><br /></p>
</div>
<p>Note that the instantaneous rates are only entered into the off-diagonal parts of the matrix. Along the diagonal, these matrices always have a set of negative numbers. For any <span class="math inline"><strong>Q</strong></span> matrix, the sum of all the elements in each row is zero – a necessary condition for a transition rate matrix. Because of this, each negative number has a value, <span class="math inline"><em>d</em><sub><em>i</em></sub></span>, equal to the sum of all of the other numbers in the row. For example,</p>
(eq. 7.2)
<div>
<p><br /><span class="math display">$$
d_1 = \sum_{i=2}^{k} q_{1i}
$$</span><br /></p>
</div>
<p>For a two-state Mk model, <span class="math inline"><em>k</em> = 2</span> and rates are symmetric so that <span class="math inline"><em>q</em><sub>12</sub> = <em>q</em><sub>21</sub></span>. In this case, we can write the transition rate matrix as:</p>
(eq. 7.3)
<div>
<p><br /><span class="math display">$$
\mathbf{Q} =
\begin{bmatrix}
-q &amp; q \\
q &amp; -q \\
\end{bmatrix}
$$</span><br /></p>
</div>
<p>Likewise, for <span class="math inline"><em>k</em> = 3</span>,</p>
<p>(eq. 7.4)</p>
<div>
<p><br /><span class="math display">$$
\mathbf{Q} =
\begin{bmatrix}
-2 q &amp; q &amp; q \\
q &amp; -2 q &amp; q \\
q &amp; q &amp; -2 q \\
\end{bmatrix}
$$</span><br /></p>
</div>
<p>In general,</p>
<p>(eq. 7.5)</p>
<div>
<p><br /><span class="math display">$$
\mathbf{Q} = q
\begin{bmatrix}
1-k &amp; 1 &amp; \dots &amp; 1 \\
1 &amp; 1-k &amp; \dots &amp; 1 \\
\vdots &amp; \vdots &amp; \ddots &amp; \vdots\\
1 &amp; 1 &amp; \dots &amp; 1\\
\end{bmatrix}
$$</span><br /></p>
</div>
<p>Once we have this transition rate matrix, we can calculate the probability distribution of trait states after any time interval <span class="math inline"><em>t</em></span> using the equation <span class="citation">(Lewis <a href="#ref-Lewis2001-bu">2001</a>)</span>:</p>
(eq. 7.6)
<div>
<p><br /><span class="math display"><strong>P</strong>(<em>t</em>)=<em>e</em><sup><strong>Q</strong><em>t</em></sup></span><br /></p>
</div>
<p>This equation looks simple, but calculating <span class="math inline"><em>P</em>(<em>t</em>)</span> involves matrix exponentiation – raising <span class="math inline"><em>e</em></span> to a power defined by a matrix. This calculation is substantially different from raising <span class="math inline"><em>e</em></span> to the power defined by each element of a matrix. I will not cover the details of matrix exponentiation here – interested readers should see Yang (2006) for details – but the calculations are not trivial. The result is a matrix, <span class="math inline"><strong>P</strong></span>, of transition probabilities. Each element in this matrix (<span class="math inline"><em>p</em><sub><em>i</em><em>j</em></sub></span>) gives the probability that starting in state <span class="math inline"><em>i</em></span> you will end up in state <span class="math inline"><em>j</em></span> over that time interval <span class="math inline"><em>t</em></span>. For the standard Mk model, there is a general solution to this equation:</p>
<p>(eq. 7.7)</p>
<div>
<p><br /><span class="math display">$$
\begin{array}{l}
p_{ii}(t) = \frac{1}{k} + \frac{k-1}{k} e^{-kqt} \\
p_{ij}(t) = \frac{1}{k} - \frac{1}{k} e^{-kqt} \\
\end{array}
$$</span><br /></p>
</div>
<p>In particular, when k = 2,</p>
<p>(eq. 7.8)</p>
<div>
<p><br /><span class="math display">$$
\begin{array}{l}
p_{ii}(t) = \frac{1}{k} + \frac{k-1}{k} e^{-kqt} = \frac{1}{2} + \frac{2-1}{2}e^{-2qt}=\frac{1+e^{-2qt}}{2} \\
p_{ij}(t) = \frac{1}{k} - \frac{1}{k} e^{-kqt} = \frac{1}{2} - \frac{1}{2}e^{-2qt}=\frac{1-e^{-2qt}}{2} \\
\end{array}
$$</span><br /></p>
</div>
<p>If we consider what happens when time gets very large in these equations, we see an interesting pattern. Any term that has <span class="math inline"><em>e</em><sup>−<em>t</em></sup></span> in it gets closer and closer to zero as t increases. Because of this, for all values of <span class="math inline"><em>k</em></span>, each <span class="math inline"><em>p</em><sub><em>i</em><em>j</em></sub>(<em>t</em>)</span> converges to a constant value, <span class="math inline">1/<em>k</em></span>. This is the stationary distribution of character states, <span class="math inline"><em>π</em></span>, defined as the equilibrium frequency of character states if the process is run many times for a long enough time period. In general, the stationary distribution of an Mk model is:</p>
(eq. 7.9)
<div>
<p><br /><span class="math display">$$
\pi =
\begin{bmatrix}
1/k &amp; 1/k &amp; \dots &amp; 1/k\\
\end{bmatrix}
$$</span><br /></p>
</div>
<p>In the case of k = 2,</p>
(eq. 7.10)
<div>
<p><br /><span class="math display">$$
\pi =
\begin{bmatrix}
1/2 &amp; 1/2 \\
\end{bmatrix}
$$</span><br /></p>
</div>
<h2 id="the-extended-mk-model">The Extended Mk Model</h2>
<p>The Mk model assumes that transitions among all possible character states occur at the same rate. However, that may not be a valid assumption. For example, it is often supposed that it is easier to lose a complex character than to gain one. We might want to fit models that allow for such asymmetries in rates.</p>
<p>For models of DNA sequence evolution there are a wide range of models allowing different rates between distinct types of nucleotides <span class="citation">(Yang <a href="#ref-Yang2006-md">2006</a>)</span>. Unequal rates are usually incorporated into the Mk model in two ways. First, one can consider the symmetric model <span class="citation">(SYM; Paradis et al. <a href="#ref-Paradis2004-xk">2004</a>)</span>. In the symmetric model, the rate of change between any two character states is the same forwards as it is backwards (that is, rates of change are symmetric; <span class="math inline"><em>q</em><sub><em>i</em><em>j</em></sub> = <em>q</em><sub><em>j</em><em>i</em></sub></span>). The rate for a particular pair of states might differ from other pairs of character stats. The rate matrix for this model has as many free rate parameters as there are pairs of character states, <span class="math inline"><em>k</em>(<em>k</em> − 1)/2</span>.</p>
(eq. 7.11)
<div>
<p><br /><span class="math display">$$
p = \frac{k(k-1)}{2}
$$</span><br /></p>
</div>
<p>However, in general symmetric models will not have stationary distributions where all character states occur at equal frequencies, as noted above for the Mk model. We can account for these uneven frequencies by adding additional parameters to our model:</p>
(eq. 7.12)
<div>
<p><br /><span class="math display">$$
\pi_{SYM} =
\begin{bmatrix}
\pi_1 &amp; \pi_2 &amp; \dots &amp; 1 - \sum_{i=1}^{n-1} \pi_i
\end{bmatrix}
$$</span><br /></p>
</div>
<p>Note that we only have to specify <span class="math inline"><em>n</em> − 1</span> equilibrium frequencies, since we know that they all sum to one. We have added <span class="math inline"><em>n</em> − 1</span> new parameters, for a total number of parameters:</p>
(eq. 7.13)
<div>
<p><br /><span class="math display">$$
p = \frac{k(k-1)}{2} + n-1
$$</span><br /></p>
</div>
<p>To obtain a <span class="math inline"><strong>Q</strong></span>-matrix for this model, we combine the information from both the relative transition rates and equilibrium frequencies:</p>
(eq. 7.14)
<div>
<p><br /><span class="math display">$$
\mathbf{Q} =
\begin{bmatrix}
\cdot &amp; r_1 &amp; \dots &amp; r_{n-1} \\
r_1 &amp; \cdot &amp; \dots &amp; \vdots \\
\vdots &amp; \vdots &amp; \cdot &amp; r_{k(k-1)/2} \\
r_{n-1} &amp; \dots r_{k(k-1)/2} &amp; \cdot \\
\end{bmatrix}
\begin{bmatrix}
\pi_1 &amp; 0 &amp; 0 &amp; 0 \\
0 &amp; \pi_2 &amp; 0 &amp; 0 \\
0 &amp; 0 &amp; \ddots &amp; 0 \\
0 &amp; 0 &amp; 0 &amp; \pi_n \\
\end{bmatrix}
$$</span><br /></p>
</div>
<p>In this equation I have left the diagonal of the first matrix as dots. The final <span class="math inline"><strong>Q</strong></span>-matrix must have all rows sum to one, so one can adjust the values of that matrix after the multiplication step.</p>
<p>In the case of a two-state model, for example, we can create a model where the forward rate is double the backward rate, and the equilibrium frequency of character one is 0.75. Then:</p>
(eq. 7.15)
<div>
<p><br /><span class="math display">$$
\mathbf{Q} =
\begin{bmatrix}
\cdot &amp; 1 \\
2 &amp; \cdot \\
\end{bmatrix}
\begin{bmatrix}
0.75 &amp; 0 \\
0 &amp; 0.25 \\
\end{bmatrix}
=
\begin{bmatrix}
\cdot &amp; 0.25 \\
1.5 &amp; \cdot \\
\end{bmatrix}
=
\begin{bmatrix}
-0.25 &amp; 0.25 \\
1.5 &amp; -1.5 \\
\end{bmatrix}
$$</span><br /></p>
</div>
<p>It is worth noting that this approach of setting parameters that define equilibrium state frequences, although borrowed from molecular evolution, is not completely standard in the comparative methods literature. One also sees equilibrium frequencies treated as a fixed property of the model, and assumed to be either equal across states or tied directly to the parameters in the -matrix.</p>
<p>The second common extension of the Mk model is called the all-rates-different model <span class="citation">(ARD; Paradis et al. <a href="#ref-Paradis2004-xk">2004</a>)</span>. In this model every possible type of transition can have a different rate. There are thus <span class="math inline"><em>k</em>(<em>k</em> − 1)</span> free rate parameters for this model, and again <span class="math inline"><em>n</em> − 1</span> parameters to specify the equilibrium frequencies of the character states.</p>
<p>The same algorithm can be used to calculate the likelihood for both of these extended Mk models (SYM and ARD). These models have more parameters than the standard Mk. To find maximum likelihood solutions, we must optimize the likelihood across the entire set of unknown parameters (see Chapter 7).</p>
<h2 id="simulating-the-mk-model-on-a-tree">Simulating the Mk model on a tree</h2>
<p>We can also use the equations above to simulate evolution under an Mk or extended-Mk model on a tree <span class="citation">(Felsenstein <a href="#ref-Felsenstein2004-eo">2004</a>)</span>. To do this, we simulate character evolution on each branch of the tree, starting at the root and progressing towards the tips. At speciation, we assume that both daughter species inherit the character state of their parental species immediately following speciation, and then evolve independently after that. At the end of the simulation, we will obtain a set of character states, one for each tip in the tree. The distribution of character states will depend on the shape of the phylogenetic tree (both its topology and branch lengths) along with the parameters of our model of character evolution.</p>
<p>We first draw a beginning character state at the root of the tree. There are several common ways to do this. For example, we can either draw from the stationary distribution or from one where each character state is equally likely. In the case of the standard Mk model, these are the same. For example, if we are simulating evolution under Mk with <span class="math inline"><em>k</em> = 2</span>, then state 0 and 1 each have a probability of <span class="math inline">1/2</span> at the root. We can draw the root state from a binomial distribution with <span class="math inline"><em>p</em><sub><em>s</em><em>t</em><em>a</em><em>t</em><em>e</em><sub>0</sub></sub> = 0.5</span>.</p>
<p>Once we have a character state for the root, we then simulate evolution along each branch in the tree. We start with the (usually two) branches descending from the root. We then proceed up the tree, branch by branch, until we get to the tips.</p>
<p>We can understand this algorithm perfectly well by thinking about what happens on each branch of the tree, and then extending that algorithm to all of the branches (as described above). For each branch, we first calculate <span class="math inline"><strong>P</strong>(<em>t</em>)</span>, the transition probability matrix, given the length of the branch and our model of evolution as summarized by <span class="math inline"><strong>Q</strong></span> and the branch length <span class="math inline"><em>t</em></span>. We then focus on the row of <span class="math inline"><strong>P</strong>(<em>t</em>)</span> that corresponds to the character state at the beginning of the branch. For example, let’s consider a basic two-state Mk model with <span class="math inline"><em>q</em> = 0.5</span>. We will call the states 0 and 1. We can calculate <span class="math inline"><strong>P</strong>(<em>t</em>)</span> for a branch with length <span class="math inline"><em>t</em> = 3</span> as:</p>
(eq. 7.16)
<div>
<p><br /><span class="math display">$$
\mathbf{P}(t) = e^{\mathbf{Q} t} = exp(
\begin{bmatrix}
-0.5 &amp; 0.5 \\
0.5 &amp; -0.5 \\
\end{bmatrix}
\cdot 3) =
\begin{bmatrix}
0.525 &amp; 0.475 \\
0.475 &amp; 0.525 \\
\end{bmatrix}
$$</span><br /></p>
</div>
<p>If we had started with character state 0 at the beginning of this branch, we would focus on the first row of this matrix. We want to end up at state 0 with probability 0.525 and change to state 1 with probability 0.475. We again draw a uniform random deviate <span class="math inline"><em>u</em></span>, and choose state 0 if <span class="math inline">0 ≤ <em>u</em> &lt; 0.525</span> and state 1 if <span class="math inline">0.525 ≤ <em>u</em> &lt; 1</span>. If we started with a different character state, we would use a different row in the matrix. If this is an internal branch in the tree, then both daughter species inherit the character state that we chose immediately following speciation – but might diverge soon after! By repeating this along every branch in the tree, we obtain a set of character states at the tips of the tree. This is then the output of our simulation.</p>
<p>Two additional details here are worth noting. First, the procedure for simulating characters under the extended-Mk model are identical to those above, except that the calculation of matrix exponentials is more complicated than in the standard Mk model. Second, if you are simulating a character with more than two states, then the procedure for drawing a random number is slightly different. One still obtains the relevant row from <span class="math inline"><strong>P</strong>(<em>t</em>)</span> and draws a uniform random deviate <span class="math inline"><em>u</em></span>. Imagine that we have a ten-state character with states 0 - 9. We start at state 0 at the beginning of the simulation. Again using <span class="math inline"><em>q</em> = 0.5</span> and <span class="math inline"><em>t</em> = 3</span>, we find that:</p>
(eq. 7.17)
<div>
<p><br /><span class="math display">$$
\begin{array}{l}
p_{ii}(t) = \frac{1}{k} + \frac{k-1}{k} e^{-kqt} = \frac{1}{10}+\frac{9}{10}e^{-2 \cdot 0.5 \cdot 3} = 0.145\\
p_{ij}(t) = \frac{1}{k} - \frac{1}{k} e^{-kqt} \frac{1}{10} - \frac{1}{10}e^{-2 \cdot 0.5 \cdot 3} = 0.095\\
\end{array}
$$</span><br /></p>
</div>
<p>We focus on the first row of <span class="math inline"><strong>P</strong>(<em>t</em>)</span>, which has elements:</p>
<div>
<p><br /><span class="math display">$$
\begin{bmatrix}
0.145 &amp; 0.095 &amp; 0.095 &amp; 0.095 &amp; 0.095 &amp; 0.095 &amp; 0.095 &amp; 0.095 &amp; 0.095 &amp; 0.095 \\
\end{bmatrix}
$$</span><br /></p>
</div>
<p>We calculate the cumulative sum of these elements, adding them together so that each number represents the sum of itself and all preceding elements in the vector:</p>
<div>
<p><br /><span class="math display">$$
\begin{bmatrix}
0.145 &amp; 0.240 &amp; 0.335 &amp; 0.430 &amp; 0.525 &amp; 0.620 &amp; 0.715 &amp; 0.810 &amp; 0.905 &amp; 1.000 \\
\end{bmatrix}
$$</span><br /></p>
</div>
<p>Now we compare <span class="math inline"><em>u</em></span> to the numbers in this cumulative sum vector. We select the smallest element that is still strictly larger than <span class="math inline"><em>u</em></span>, and assign this character state for the end of the branch. For example, if <span class="math inline"><em>u</em> = 0.475</span>, the 5th element, 0.525, is the smallest number that is still greater than <span class="math inline"><em>u</em></span>. This corresponds to character state 4, which we assign to the end of the branch. This last procedure is a numerical trick. Imagine that we have a line segment with length 1. The cumulative sum vector breaks the unit line into segments, each of which is exactly as long as the probability of each event in the set. One then just draws a random number between 0 and 1 using a uniform distribution. The segment that contains this random number is our event.</p>
<p>We can apply this approach to simulate the evolution of limblessness in squamates. Below, I present the results of three such simulations. These simulations are a little different than what I describe above because they consider all changes in the tree, rather than just character states at nodes and tips; but the model (and the principal) is the same. You can see that the model leaves an imprint on the pattern of changes in the tree, and you can imagine that one might be able to reconstruct the model using a phylogenetic comparative approach. Of course, typically we know only the tip states, and have to reconstruct changes along branches in the tree. We will discuss parameter estimation for the Mk and extended-Mk models in the next chapter.</p>
<div class="figure">
<img src="{{site.baseurl}}/images/figure7-4.png" alt="Figure 7.4. Simulated character evolution on a phylogenetic tree of squamates from(from Brandley et al. 2008) under an equal-rates Mk model with slow, fast, and asymmetric transition rates (from right to left). In all three cases, I assumed that the ancestor of squamates had limbs." />
<p class="caption">Figure 7.4. Simulated character evolution on a phylogenetic tree of squamates <span class="citation">(from Brandley et al. <a href="#ref-Brandley2008-wr">2008</a>)</span> under an equal-rates Mk model with slow, fast, and asymmetric transition rates (from right to left). In all three cases, I assumed that the ancestor of squamates had limbs.</p>
</div>
<h2 id="chapter-summary">Chapter summary</h2>
<!---
<img data-gifffer="{{ site.baseurl }}/images/figure7-animated.png" />
-->
<p>In this chapter I have described the Mk model, which can be used to describe the evolution of discrete characters that have a set number of fixed states. We can also elaborate on the Mk model to allow more complex models of discrete character evolution (the extended-Mk model). These models can all be used to simulate the evolution of discrete characters on trees.</p>
<p>In summary, the Mk and extended Mk model are general models that one can use for the evolution of discrete characters. In the next chapter, I will show how to fit these models to data and use them to test evolutionary hypotheses.</p>
<h2 id="references" class="unnumbered">References</h2>
<div id="refs" class="references">
<div id="ref-Brandley2008-wr">
<p>Brandley, M. C., J. P. Huelsenbeck, and J. J. Wiens. 2008. Rates and patterns in the evolution of snake-like body form in squamate reptiles: Evidence for repeated re-evolution of lost digits and long-term persistence of intermediate body forms. Evolution. Wiley Online Library.</p>
</div>
<div id="ref-Felsenstein2004-eo">
<p>Felsenstein, J. 2004. Inferring phylogenies. Sinauer Associates, Inc., Sunderland, MA.</p>
</div>
<div id="ref-Lewis2001-bu">
<p>Lewis, P. O. 2001. A likelihood approach to estimating phylogeny from discrete morphological character data. Syst. Biol. 50:913–925.</p>
</div>
<div id="ref-Moch2011-fp">
<p>Moch, J. G., and P. Senter. 2011. Vestigial structures in the appendicular skeletons of eight african skink species (squamata, scincidae). J. Zool. 285:274–280. Blackwell Publishing Ltd.</p>
</div>
<div id="ref-Pagel1994-ui">
<p>Pagel, M. 1994. Detecting correlated evolution on phylogenies: A general method for the comparative analysis of discrete characters. Proc. R. Soc. Lond. B Biol. Sci. 255:37–45. The Royal Society.</p>
</div>
<div id="ref-Paradis2004-xk">
<p>Paradis, E., J. Claude, and K. Strimmer. 2004. APE: Analyses of phylogenetics and evolution in R language. Bioinformatics 20:289–290. academic.oup.com.</p>
</div>
<div id="ref-Pianka2017-aw">
<p>Pianka, E. R., L. J. Vitt, N. Pelegrin, D. B. Fitzgerald, and K. O. Winemiller. 2017. Toward a periodic table of niches, or exploring the lizard niche hypervolume. Am. Nat. 190:601–616. journals.uchicago.edu.</p>
</div>
<div id="ref-Posada2008-qw">
<p>Posada, D. 2008. jModelTest: Phylogenetic model averaging. Mol. Biol. Evol. 25:1253–1256. academic.oup.com.</p>
</div>
<div id="ref-Streicher2017-wb">
<p>Streicher, J. W., and J. J. Wiens. 2017. Phylogenomic analyses of more than 4000 nuclear loci resolve the origin of snakes among lizard families. Biol. Lett. 13. rsbl.royalsocietypublishing.org.</p>
</div>
<div id="ref-Vitt2003-bl">
<p>Vitt, L. J., E. R. Pianka, W. E. Cooper Jr, and K. Schwenk. 2003. History and the global ecology of squamate reptiles. Am. Nat. 162:44–60. journals.uchicago.edu.</p>
</div>
<div id="ref-Yang2006-md">
<p>Yang, Z. 2006. Computational molecular evolution. Oxford University Press.</p>
</div>
</div>
</body>
</html>
