---
layout: chapter
title: Characters and diversification rates
---
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
  <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
  <meta http-equiv="Content-Style-Type" content="text/css" />
  <meta name="generator" content="pandoc" />
  <title></title>
  <style type="text/css">code{white-space: pre;}</style>
</head>
<body>
<div id="TOC">
<ul>
<li><a href="#chapter-13-characters-and-diversification-rates">Chapter 13: Characters and diversification rates</a><ul>
<li><a href="#section-13.1-the-evolution-of-self-incompatibility">Section 13.1: The evolution of self-incompatibility</a></li>
<li><a href="#section-13.2-a-state-dependent-model-of-diversification">Section 13.2: A State-Dependent Model of Diversification</a></li>
<li><a href="#section-13.3-calculating-likelihoods-for-state-dependent-diversification-models">Section 13.3: Calculating Likelihoods for State-dependent diversification models</a></li>
<li><a href="#section-13.4-ml-and-bayesian-tests-for-state-dependent-diversification">Section 13.4: ML and Bayesian Tests for State-Dependent Diversification</a></li>
<li><a href="#section-13.5potential-pitfalls-and-how-to-avoid-them">Section 13.5:Potential Pitfalls and How to Avoid Them</a></li>
<li><a href="#section-13.4-summary">Section 13.4: Summary</a></li>
<li><a href="#footnotes">Footnotes</a></li>
</ul></li>
<li><a href="#references">References</a></li>
</ul>
</div>
<div>
<p><a href="{{site.baseurl}}/pdf/chapter13_chardiv.pdf">pdf version</a></p>
</div>
<h1 id="chapter-13-characters-and-diversification-rates">Chapter 13: Characters and diversification rates</h1>
<h2 id="section-13.1-the-evolution-of-self-incompatibility">Section 13.1: The evolution of self-incompatibility</h2>
<p>Most people have not spent a lot of time thinking about the sex lives of plants. The classic mode of sexual reproduction in angiosperms (flowering plants) involves pollen (the male gametophyte stage of the plant life cycle). Pollen lands on the pistil (the female reproductive structure) and produces a pollen tube. Sperm cells move down the pollen tube, and one sperm cell unites with the egg to form a new zygote in the ovule.</p>
<p>As you might imagine, plants have little control over what pollen grains land on their pistil <span class="citation">(although plant species do have some remarkable adaptations to control pollination by animals; see Anders Nilsson <a href="#ref-Anders_Nilsson1992-af">1992</a>)</span>. In particular, this mode of reproduction leaves open the possibility of self-pollination, where pollen from a plant fertilizes eggs from the same plant <span class="citation">(Stebbins <a href="#ref-Stebbins1950-aj">1950</a>)</span>. Self-fertilization (sometimes called selfing) is a form of asexual reproduction, but one that involves meiosis; as such, there are costs to self-fertilization. The main cost is inbreeding depression, a reduction in offspring fitness associated with recessive deleterious alleles across the genome <span class="citation">(Holsinger et al. <a href="#ref-Holsinger1984-ts">1984</a>)</span>.</p>
<p>Some species of angiosperms can avoid self-fertilization through self-incompatibility <span class="citation">(Bateman <a href="#ref-Bateman1952-iy">1952</a>)</span>. In plants with self-incompatibility, the process by which the sperm meets the egg is interrupted at some stage if pollen grains have a genotype that is the same as the parent <span class="citation">(e.g. Schopfer et al. <a href="#ref-Schopfer1999-cm">1999</a>)</span>. This prevents selfing – and also prevents sexual reproduction with plants that have the same genotype(s) at loci involved in the process.</p>
<p>Species of angiosperms are about evenly divided between these two states of self-compatibility and self-incompatibility <span class="citation">(Igic and Kohn <a href="#ref-Igic2006-zg">2006</a>)</span>. Furthermore, self-incompatible species are scattered throughout the phylogenetic tree of angiosperms <span class="citation">(Igic and Kohn <a href="#ref-Igic2006-zg">2006</a>)</span>.</p>
<p>The evolution of selfing is a good example of a trait that might have a strong effect on diversification rates. One can easily imagine, for example, how incompatibility loci might facilitate the evolution of reproductive isolation among populations, and how lineages with such loci might diversity at a very different tempo than those without <span class="citation">(Goldberg et al. <a href="#ref-Goldberg2010-ua">2010</a>)</span>.</p>
<p>In this chapter, we will learn about a family of models where traits can affect diversification rates. I will also address some of the controversial aspects of these models and how we can improve these approaches in the future.</p>
<h2 id="section-13.2-a-state-dependent-model-of-diversification">Section 13.2: A State-Dependent Model of Diversification</h2>
<p>The models that we will consider in this chapter include both a model of trait evolution and an associated model of lineage diversification. In the simplest case, we can consider a model where the character has two states, 0 and 1. We need to model the transitions among these states, which we can do in an identical way to what we did in Chapter 7 using a continuous-time Markov model. We express this model using two rate parameters, a forward rate <span class="math inline"><em>q</em><sub>01</sub></span> and a backwards rate <span class="math inline"><em>q</em><sub>10</sub></span>.</p>
<p>We now consider the idea that diversification rates might depend on the character state. We assume that species with character state 0 have a certain speciation rate (<span class="math inline"><em>λ</em><sub>0</sub></span>) and extinction rate (<span class="math inline"><em>μ</em><sub>0</sub></span>), and that species in 1 have potentially different rates of both speciation (<span class="math inline"><em>λ</em><sub>1</sub></span>) and extinction (<span class="math inline"><em>μ</em><sub>1</sub></span>). Thus, we have a six-parameter model <span class="citation">(Maddison et al. <a href="#ref-Maddison2007-vu">2007</a>)</span>. We assume that parent lineages give birth to daughters with the same character state, that is that character states do not change at speciation.</p>
<p>It is straightforward to simulate evolution under our state-dependent model of diversification. We proceed in the same way as we did for birth-death models, by drawing waiting times, but these waiting times can be waiting times to the next character state change, speciation, or extinction event. In particular, imagine that there are <span class="math inline"><em>n</em></span> lineages present at time <span class="math inline"><em>t</em></span>, and that <span class="math inline"><em>k</em></span> of these lineages are in state 0 (and <span class="math inline"><em>n</em> − <em>k</em></span> are in state 1). The waiting time to the next event will follow an exponential distribution with a rate parameter of:</p>
<p>(eq. 13.1)</p>
<div>
<p><br /><span class="math display"><em>ρ</em> = <em>n</em>(<em>q</em><sub>01</sub> + <em>λ</em><sub>0</sub> + <em>μ</em><sub>0</sub>)+(<em>k</em> − <em>n</em>)(<em>q</em><sub>10</sub> + <em>λ</em><sub>1</sub> + <em>μ</em><sub>1</sub>)</span><br /></p>
</div>
<p>Once we have a waiting time, we can assign an event type depending on probabilities. For example, the probability that the event is a character state change from 0 to 1 is:</p>
<p>(eq. 13.2)</p>
<div>
<p><br /><span class="math display"><em>p</em><sub><em>q</em><sub>01</sub></sub> = (<em>n</em> ⋅ <em>q</em><sub>01</sub>)/<em>ρ</em></span><br /></p>
</div>
<p>And the probability that the event is the extinction of a lineage with character state 1 is:</p>
<p>(eq. 13.3)</p>
<div>
<p><br /><span class="math display"><em>p</em><sub><em>μ</em><sub>1</sub></sub> = [(<em>n</em> − <em>k</em>)⋅<em>μ</em><sub>1</sub>]/<em>ρ</em></span><br /></p>
</div>
<p>And so on for the other four possible events.</p>
<p>Once we have picked an event in this way, we can randomly assign it to one of the lineages in the appropriate set, with each lineage equally likely to be chosen. We then proceed forwards in time until we have a dataset with the desired size or total time depth.</p>
<p>An example simulation is shown in Figure 14.1. As you can see, under these model parameters the impact of character states on diversification is readily apparent. In the next section we will figure out how to extract that information from our data.</p>
<div class="figure">
<img src="{{site.baseurl}}/images/figure13-1.png" alt="Figure 13.1. Simulation of character-dependent diversification. Data were simulated under a model where diversification rate of state zero (red) is substantially lower than that of state 1 (black; model parameters q_{01} = 1_{10} = 0.05, \lambda_0 = 0.2, \lambda_1 = 0.8, \mu_0 = \mu_1 = 0.05)." />
<p class="caption">Figure 13.1. Simulation of character-dependent diversification. Data were simulated under a model where diversification rate of state zero (red) is substantially lower than that of state 1 (black; model parameters <span class="math inline"><em>q</em><sub>01</sub> = 1<sub>10</sub> = 0.05</span>, <span class="math inline"><em>λ</em><sub>0</sub> = 0.2</span>, <span class="math inline"><em>λ</em><sub>1</sub> = 0.8</span>, <span class="math inline"><em>μ</em><sub>0</sub> = <em>μ</em><sub>1</sub> = 0.05</span>).</p>
</div>
<h2 id="section-13.3-calculating-likelihoods-for-state-dependent-diversification-models">Section 13.3: Calculating Likelihoods for State-dependent diversification models</h2>
<p>To calculate likelihoods for state-dependent diversification models we use a pruning algorithm with calculations that progress back through the tree from the tips to the root. We will follow the description of this algorithm in Maddison et al. <span class="citation">(<a href="#ref-Maddison2007-vu">2007</a>)</span>. We have already used this paper to derive likelihoods for constant rate birth-death models on trees (Chapter 12), and this derivation is closely related that previous approach.</p>
<p>We consider a phylogenetic tree with data on the character states at the tips. For the purposes of this example, we will assume that the tree is complete and correct – we are not missing any species, and there is no phylogenetic uncertainty. We will come back to these two assumptions later in the chapter.</p>
<p>We need to obtain the probability of obtaining the data given the model (the likelihood). As we have seen before, we will calculate that likelihood going backwards in time using a pruning algorithm <span class="citation">(Maddison et al. <a href="#ref-Maddison2007-vu">2007</a>)</span>. The key principle, again, is that if we know the probabilities at some point in time on the tree, we can calculate those probabilities at some time point immediately before. By applying this method successively, we can move back towards the root of the tree. We move backwards along each branch in the tree, merging these calculations at nodes. When we get to the root, we have the probability of the data given the model and the entire tree – that is, we have the likelihood.</p>
<p>The other essential piece is that we have a starting point. When we start at the tips of the tree, we assume that our character states are fixed and known. We use the fact that we know all of the species and their character states at the present day as our starting point, and move backwards from there <span class="citation">(Maddison et al. <a href="#ref-Maddison2007-vu">2007</a>)</span>. For example, for a species with character state 0, the likelihood for state 0 is 1, and for state 1 is zero. In other words, at the tips of the tree we can start our calculations with a probability of 1 for the state that matches the tip state, and 0 otherwise.</p>
<p>This discussion also highlights the fact that incorporating uncertainty and/or variation in tip states for these algorithms is not computationally difficult – we just need to start from a different point at the tips. For example, if we are completely unsure about the tip state for a certain taxa, we can begin with likelihoods of 0.5 for starting in state 0 and 0.5 for starting in state 1. However, such calculations are not commonly implemented in comparative methods software.</p>
<p>We now need to consider the change in the likelihood as we step backwards through time in the tree <span class="citation">(Maddison et al. <a href="#ref-Maddison2007-vu">2007</a>)</span>. We will consider some very small time interval <span class="math inline"><em>Δ</em><em>t</em></span>, and later use differential equations to find out what happens in the limit as this interval goes to zero (Figure 13.2). Since we will eventually take the limit as <span class="math inline"><em>Δ</em><em>t</em> → 0</span>, we can assume that the time interval is so small that, at most, one event (speciation, extinction, or character change) has happened in that interval, but never more than one. We will calculate the probability of the observed data given that the character is in each state at time <span class="math inline"><em>t</em></span>, again measuring time backwards from the present day. In other words, we are considering the probability of the observed data if, at time t, the character state were in state 0 [<span class="math inline"><em>p</em><sub>0</sub>(<em>t</em>)</span>] or state 1 [<span class="math inline"><em>p</em><sub>1</sub>(<em>t</em>)</span>]. For now, we can assume we know these probabilities, and try to calculate updated probabilities at some earlier time <span class="math inline"><em>t</em> + <em>Δ</em><em>t</em></span>: <span class="math inline"><em>p</em><sub>0</sub>(<em>t</em> + <em>Δ</em><em>t</em>)</span> and <span class="math inline"><em>p</em><sub>1</sub>(<em>t</em> + <em>Δ</em><em>t</em>)</span>.</p>
<div class="figure">
<img src="{{site.baseurl}}/images/figure13-2.png" alt="Figure 13.2. Taken from Maddison - to be redrawn." />
<p class="caption">Figure 13.2. Taken from Maddison - to be redrawn.</p>
</div>
<p>To calculate <span class="math inline"><em>p</em><sub>0</sub>(<em>t</em> + <em>Δ</em><em>t</em>)</span> and <span class="math inline"><em>p</em><sub>1</sub>(<em>t</em> + <em>Δ</em><em>t</em>)</span>, we consider all of the possible things that could happen in a time interval <span class="math inline"><em>Δ</em><em>t</em></span> along a branch in a phylogenetic tree that are compatible with our dataset <span class="citation">(Figure 13.2; Maddison et al. <a href="#ref-Maddison2007-vu">2007</a>)</span>. First, nothing at all could have happened; second, our character state could have changed; and third, there could have been a speciation event. This last event might seem incorrect, as we are only considering changes along branches in the tree and not at nodes. If we did not reconstruct any speciation events at some point along a branch, then how could one have taken place? The answer is that a speciation event could have occurred but all taxa descended from that branch have since gone extinct. We must also consider the possibility that either the right or the left lineage went extinct following the speciation event; that is why the speciation event probabilities appear twice in Figure 13.2 <span class="citation">(Maddison et al. <a href="#ref-Maddison2007-vu">2007</a>)</span>.</p>
<div class="figure">
<img src="{{site.baseurl}}/images/figure13-3.png" alt="Figure 13.3. Taken from Maddison - to be redrawn." />
<p class="caption">Figure 13.3. Taken from Maddison - to be redrawn.</p>
</div>
<p>We can write an equation for these updated probabilities. We will consider the probability that the character is in state 0 at time <span class="math inline"><em>t</em> + <em>Δ</em><em>t</em></span>; the equation for state 1 is similar <span class="citation">(Maddison et al. <a href="#ref-Maddison2007-vu">2007</a>)</span>.</p>
<p>(eq. 13.4)</p>
<div>
<p><br /><span class="math display"><em>p</em><sub>0</sub>(<em>t</em> + <em>Δ</em><em>t</em>)=(1 − <em>μ</em><sub>0</sub>)<em>Δ</em><em>t</em> ⋅ [(1 − <em>q</em><sub>01</sub><em>Δ</em><em>t</em>)(1 − <em>λ</em><sub>0</sub><em>Δ</em><em>t</em>)<em>p</em><sub>0</sub>(<em>t</em>)+<em>q</em><sub>01</sub><em>Δ</em><em>t</em>(1 − <em>λ</em><sub>0</sub><em>Δ</em><em>t</em>)<em>p</em><sub>1</sub>(<em>t</em>)+2 ⋅ (1 − <em>q</em><sub>01</sub><em>Δ</em><em>t</em>)<em>λ</em><sub>0</sub><em>Δ</em><em>t</em> ⋅ <em>E</em><sub>0</sub>(<em>t</em>)<em>p</em><sub>0</sub>(<em>t</em>)]</span><br /></p>
</div>
<p>We can multiply through and simplify. We will also drop any terms that include <span class="math inline">[<em>Δ</em><em>t</em>]<sup>2</sup></span>, which become vanishingly small as <span class="math inline"><em>Δ</em><em>t</em></span> decreases. Doing that, we obtain <span class="citation">(Maddison et al. <a href="#ref-Maddison2007-vu">2007</a>)</span>:</p>
<p>(eq. 13.5)</p>
<div>
<p><br /><span class="math display"><em>p</em><sub>0</sub>(<em>t</em> + <em>Δ</em><em>t</em>)=[1 − (<em>λ</em><sub>0</sub> + <em>μ</em><sub>0</sub> + <em>q</em><sub>01</sub>)<em>Δ</em><em>t</em>]<em>p</em><sub>0</sub>(<em>t</em>)+(<em>q</em><sub>01</sub><em>Δ</em><em>t</em>)<em>p</em><sub>1</sub>(<em>t</em>)+2(<em>λ</em><sub>0</sub><em>Δ</em><em>t</em>)<em>E</em><sub>0</sub>(<em>t</em>)<em>p</em><sub>0</sub>(<em>t</em>)</span><br /></p>
</div>
<p>Similarly,</p>
<p>(eq. 13.6)</p>
<div>
<p><br /><span class="math display"><em>p</em><sub>1</sub>(<em>t</em> + <em>Δ</em><em>t</em>)=[1 − (<em>λ</em><sub>1</sub> + <em>μ</em><sub>1</sub> + <em>q</em><sub>10</sub>)<em>Δ</em><em>t</em>]<em>p</em><sub>1</sub>(<em>t</em>)+(<em>q</em><sub>10</sub><em>Δ</em><em>t</em>)<em>p</em><sub>0</sub>(<em>t</em>)+2(<em>λ</em><sub>1</sub><em>Δ</em><em>t</em>)<em>E</em><sub>1</sub>(<em>t</em>)<em>p</em><sub>1</sub>(<em>t</em>)</span><br /></p>
</div>
<p>We can then find the rate of change for these two equations by solving for <span class="math inline"><em>p</em><sub>1</sub>(<em>t</em> + <em>Δ</em><em>t</em>)/[<em>Δ</em><em>t</em>]</span>, then taking the limit as <span class="math inline"><em>Δ</em><em>t</em> → 0</span>. This gives <span class="citation">(Maddison et al. <a href="#ref-Maddison2007-vu">2007</a>)</span>:</p>
<p>(eq. 13.7)</p>
<div>
<p><br /><span class="math display">$$
\frac{dp_0}{dt} = -(\lambda_0 + \mu_0 + q_{01}) p_0(t) + q{01}p_1(t) + 2 \lambda_0 E_0(t) p_0(t)
$$</span><br /></p>
</div>
<p>and:</p>
<p>(eq. 13.8)</p>
<div>
<p><br /><span class="math display">$$
\frac{dp_1}{dt} = -(\lambda_1 + \mu_1 + q_{10}) p_1(t) + q{10}p_1(t) + 2 \lambda_1 E_1(t) p_1(t)
$$</span><br /></p>
</div>
<p>We also need to consider <span class="math inline"><em>E</em><sub>0</sub>(<em>t</em>)</span> and <span class="math inline"><em>E</em><sub>1</sub>(<em>t</em>)</span>. These represent the probability that a lineage with state 0 or 1, respectively, and alive at time <span class="math inline"><em>t</em></span> will go extinct before the present day. Neglecting the derivation of these formulas, which can be found in Maddison et al. <span class="citation">(<a href="#ref-Maddison2007-vu">2007</a>)</span> and is closely related to similar terms in Chapters 11 and 12, we have:</p>
<p>(eq. 13.7)</p>
<div>
<p><br /><span class="math display">$$
\frac{dE_0}{dt} = \mu_0-(\lambda_0+\mu_0+q_{01} ) E_0 (t)+q_{01} E_1 (t)+\lambda_0 [E_0 (t)]^2
$$</span><br /></p>
</div>
<p>and:</p>
<p>(eq. 13.8)</p>
<div>
<p><br /><span class="math display">$$
\frac{dE_1}{dt} = \mu_1-(\lambda_1+\mu_1+q_{10} ) E_1 (t)+q_{10} E_0 (t)+\lambda_1 [E_1 (t)]^2
$$</span><br /></p>
</div>
<p>Along a single branch in a tree, we can sum together many such small time intervals. But what happens when we get to a node? Well, if we consider the time interval that contains the node, then we already know what happened – a speciation event. We also know that the two daughters immediately after the speciation event were identical in their traits (this is an assumption of the model). So we can calculate the likelihood for their ancestor for each state as the product of the likelihoods of the two daughter branches coming into that node <span class="citation">(Maddison et al. <a href="#ref-Maddison2007-vu">2007</a>)</span>. In this way, we merge our likelihood calculations along each branch when we get to nodes in the tree.</p>
<p>When we get to the root of the tree, we are almost done – but not quite! We have partial likelihood calculations for each character state – so we know, for example, the likelihood of the data if we had started with a root state of 0, and also if we had started at 1. To merge these we need to use probabilities of each character state at the root of the tree <span class="citation">(Maddison et al. <a href="#ref-Maddison2007-vu">2007</a>)</span>. For example, if we do not know the root state from any outside information, we might consider root probabilities for each state to be equal, 0.5 for state 0 and 0.5 for state 1. We then multiply the likelihood associated with each state with the root probability for that state. Finally, we add these likelihoods together to obtain the full likelihood of the data given the model.</p>
<p>The question of which root probabilities to use for this calculation has been discussed in the literature, and does matter in some applications. Aside from equal probabilities of each state, other options include using outside information to inform prior probabilities on each state <span class="citation">(e.g. Hagey et al. <a href="#ref-Hagey2017-hy">2017</a>)</span>, finding the calculated equilibrium frequency of each state under the model <span class="citation">(Maddison et al. <a href="#ref-Maddison2007-vu">2007</a>)</span>, or weighting each root state by its likelihood of generating the data, effectively treating the root as a nuisance parameter <span class="citation">(FitzJohn et al. <a href="#ref-FitzJohn2009-sg">2009</a>)</span>.</p>
<p>I have described the situation where we have two character states, but this method generalizes well to multi-state characters <span class="citation">(the MuSSE method; FitzJohn <a href="#ref-FitzJohn2012-tu">2012</a>)</span>. We can describe the evolution of the character in the same way as described for multi-state discrete characters in chapter 9. We then can assign unique diversification rate parameters to each of the <span class="math inline"><em>k</em></span> character states: <span class="math inline"><em>λ</em><sub>0</sub>, <em>λ</em><sub>1</sub>, …, <em>λ</em><sub><em>k</em></sub></span> and <span class="math inline"><em>μ</em><sub>0</sub>, <em>μ</em><sub>1</sub>, …, <em>μ</em><sub><em>k</em></sub></span> <span class="citation">(FitzJohn <a href="#ref-FitzJohn2012-tu">2012</a>)</span>. It is worth keeping in mind, though, that it is not too hard to construct a model where parameters are not identifiable and model fitting and estimation become very difficult.</p>
<h2 id="section-13.4-ml-and-bayesian-tests-for-state-dependent-diversification">Section 13.4: ML and Bayesian Tests for State-Dependent Diversification</h2>
<p>Now that we can calculate the likelihood for state-dependent diversification models, formulating ML and Bayesian tests follows the same pattern we have encountered before. For ML, some comparisons are nested and so you can use likelihood ratio tests. For example, we can compare the full BiSSe model <span class="citation">(Maddison et al. <a href="#ref-Maddison2007-vu">2007</a>)</span>, with parameters <span class="math inline"><em>q</em><sub>01</sub></span>, <span class="math inline"><em>q</em><sub>10</sub></span>, <span class="math inline"><em>λ</em><sub>0</sub></span>, <span class="math inline"><em>λ</em><sub>1</sub></span>, <span class="math inline"><em>μ</em><sub>0</sub></span>, <span class="math inline"><em>μ</em><sub>1</sub></span> with a restricted model with parameters <span class="math inline"><em>q</em><sub>01</sub></span>, <span class="math inline"><em>q</em><sub>10</sub></span>, <span class="math inline"><em>λ</em><sub><em>a</em><em>l</em><em>l</em></sub></span>, <span class="math inline"><em>μ</em><sub><em>a</em><em>l</em><em>l</em></sub></span>. Since the restricted model is a special case of the full model where <span class="math inline"><em>λ</em><sub>0</sub> = <em>λ</em><sub>1</sub> = <em>λ</em><sub><em>a</em><em>l</em><em>l</em></sub></span> and <span class="math inline"><em>μ</em><sub>0</sub> = <em>μ</em><sub>1</sub> = <em>μ</em><sub><em>a</em><em>l</em><em>l</em></sub></span>, we can compare the two using a likelihood ratio test, as described earlier in the book. Alternatively, we can compare a series of BiSSe-type models by comparing their <span class="math inline"><em>A</em><em>I</em><em>C</em><sub><em>c</em></sub></span> scores.</p>
<p>For example, I will apply this approach to the example of self-incompitability. I will use data from Goldberg and Igic <span class="citation">(<a href="#ref-Goldberg2012-gs">2012</a>)</span>, who provide a phylogenetic tree and data for 356 species of Solanaceae. All species were classified as having any form of self incompatibility, even if the state is variable among populations. The data, along with a stochastic character map of state changes, are shown in Figure 13.4.</p>
<div class="figure">
<img src="{{site.baseurl}}/images/figure13-4.png" alt="Figure 13.4. Data from Goldberg and Igic (2012) showing presence (red) and absence (black) of self-incomatibility among Solanaceae. Branches colored using stochastic character mapping under a model with distinct forwards and backwards rates; these reconstructions are biased if characters affect diversification rates." />
<p class="caption">Figure 13.4. Data from Goldberg and Igic <span class="citation">(<a href="#ref-Goldberg2012-gs">2012</a>)</span> showing presence (red) and absence (black) of self-incomatibility among Solanaceae. Branches colored using stochastic character mapping under a model with distinct forwards and backwards rates; these reconstructions are biased if characters affect diversification rates.</p>
</div>
<p>Applying the BiSSe models to these data and assuming that <span class="math inline"><em>q</em><sub>01</sub> ≠ <em>q</em><sub>10</sub></span>, we obtain the following results:</p>
<table style="width:28%;">
<colgroup>
<col width="5%" />
<col width="5%" />
<col width="5%" />
<col width="5%" />
<col width="5%" />
</colgroup>
<thead>
<tr class="header">
<th>Model</th>
<th>Number of parameters</th>
<th>Parameter estimates</th>
<th>lnLikelihood</th>
<th>AIC</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Character-independent model</td>
<td>4</td>
<td><span class="math inline"><em>λ</em> = 0.65</span>, <span class="math inline"><em>μ</em> = 0.57</span></td>
<td>-945.96</td>
<td>1899.9</td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td><span class="math inline"><em>q</em><sub>01</sub> = 0.16</span>, <span class="math inline"><em>q</em><sub>10</sub> = 0.09</span></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>Speciation rate depends on character</td>
<td>5</td>
<td><span class="math inline"><em>μ</em> = 0.57</span></td>
<td>-945.57</td>
<td>1901.1</td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td><span class="math inline"><em>λ</em><sub>0</sub> = 0.69</span>, <span class="math inline"><em>λ</em><sub>1</sub> = 0.63</span></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td></td>
<td></td>
<td><span class="math inline"><em>q</em><sub>01</sub> = 0.17</span>, <span class="math inline"><em>q</em><sub>10</sub> = 0.08</span></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>Extinction rate depends on character</td>
<td>5</td>
<td><span class="math inline"><em>λ</em> = 0.65</span></td>
<td>-943.93</td>
<td>1897.9</td>
</tr>
<tr class="odd">
<td></td>
<td></td>
<td><span class="math inline"><em>μ</em><sub>0</sub> = 0.45</span>, <span class="math inline"><em>μ</em><sub>1</sub> = 0.67</span></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td><span class="math inline"><em>q</em><sub>01</sub> = 0.22</span>, <span class="math inline"><em>q</em><sub>10</sub> = 0.06</span></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>Full character-dependent model</td>
<td>6</td>
<td><span class="math inline"><em>λ</em><sub>0</sub> = 0.49</span>, <span class="math inline"><em>λ</em><sub>1</sub> = 0.79</span></td>
<td>-941.94</td>
<td>1895.9</td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td><span class="math inline"><em>μ</em><sub>0</sub> = 0.20</span>, <span class="math inline"><em>μ</em><sub>1</sub> = 0.84</span></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td></td>
<td></td>
<td><span class="math inline"><em>q</em><sub>01</sub> = 0.29</span>, <span class="math inline"><em>q</em><sub>10</sub> = 0.05</span></td>
<td></td>
<td></td>
</tr>
</tbody>
</table>
<p>From this, we conclude that models where the character influences diversification fit best, with the full model receiving the most support. We can't discount the possibility that the character only influences extinction and not speciation, since that model is within 2 AIC units of the best model.</p>
<p>Alternatively, we can carry out a Bayesian test for state-dependent diversification. Like other models in the book, this requires setting up an MCMC algorithm that generates posterior distributions for our model parameters <span class="citation">(FitzJohn <a href="#ref-FitzJohn2012-tu">2012</a>)</span>. In this case:</p>
<ol style="list-style-type: decimal">
<li><p>Sample a set of starting parameter values, <span class="math inline"><em>q</em><sub>01</sub>, <em>q</em><sub>10</sub>, <em>λ</em><sub>0</sub>, <em>λ</em><sub>1</sub>, <em>μ</em><sub>0</sub>, <em>μ</em><sub>1</sub></span>, from their prior distributions. For example, one could set prior distribution for all parameters as exponential with a mean and variance of <span class="math inline"><em>λ</em><sub><em>p</em><em>r</em><em>i</em><em>o</em><em>r</em><sub><em>i</em></sub></sub></span> (note that, as usual, the choice for this parameter should depend on the units of tree branch lengths you are using). We then select starting values for all parameters from the prior.</p></li>
<li><p>Given the current parameter values, select new proposed parameter values using the proposal density <span class="math inline"><em>Q</em>(<em>p</em>′|<em>p</em>)</span>. For all parameter values, we can use a uniform proposal density with width <span class="math inline"><em>w</em><sub><em>p</em></sub></span>, so that <span class="math inline"><em>Q</em>(<em>p</em>′|<em>p</em>) <em>U</em>(<em>p</em> − <em>w</em><sub><em>p</em></sub>/2, <em>p</em> + <em>w</em><sub><em>p</em></sub>/2)</span>. We can either choose all parameter values simultaneously, or one at a time (the latter is typically more effective).</p></li>
<li><p>Calculate three ratios:</p></li>
</ol>
<ol style="list-style-type: lower-alpha">
<li>The prior odds ratio. This is the ratio of the probability of drawing the parameter values <span class="math inline"><em>p</em></span> and <span class="math inline"><em>p</em>′</span> from the prior. Since we have exponential priors for all parameters, we can calculate this ratio as:</li>
</ol>
<p>(eq. 11.5)</p>
<div>
<p><br /><span class="math display">$$
R_{prior} = \frac{\lambda_{prior_i} e^{-\lambda_{prior_i} p'}}{\lambda_{prior_i} e^{-\lambda_{prior_i} p}}=e^{\lambda_{prior_i} (p-p')}
$$</span><br /></p>
</div>
<ol start="2" style="list-style-type: lower-alpha">
<li><p>The proposal density ratio. This is the ratio of probability of proposals going from <span class="math inline"><em>p</em></span> to <span class="math inline"><em>p</em>′</span> and the reverse. We have already declared a symmetrical proposal density, so that <span class="math inline"><em>Q</em>(<em>p</em>′|<em>p</em>)=<em>Q</em>(<em>p</em>|<em>p</em>′)</span> and <span class="math inline"><em>R</em><sub><em>p</em><em>r</em><em>o</em><em>p</em><em>o</em><em>s</em><em>a</em><em>l</em></sub> = 1</span>.</p></li>
<li><p>The likelihood ratio. This is the ratio of probabilities of the data given the two different parameter values. We can calculate these probabilities from the approach described in the previous section.</p></li>
</ol>
<ol start="4" style="list-style-type: decimal">
<li>Find <span class="math inline"><em>R</em><sub><em>a</em><em>c</em><em>c</em><em>e</em><em>p</em><em>t</em></sub></span> as the product of the prior odds, proposal density ratio, and the likelihood ratio. In this case, the proposal density ratio is 1, so:</li>
</ol>
<p>(eq. 11.6)</p>
<div>
<p><br /><span class="math display"><em>R</em><sub><em>a</em><em>c</em><em>c</em><em>e</em><em>p</em><em>t</em></sub> = <em>R</em><sub><em>p</em><em>r</em><em>i</em><em>o</em><em>r</em></sub> ⋅ <em>R</em><sub><em>l</em><em>i</em><em>k</em><em>e</em><em>l</em><em>i</em><em>h</em><em>o</em><em>o</em><em>d</em></sub></span><br /></p>
</div>
<ol start="5" style="list-style-type: decimal">
<li><p>Draw a random number <span class="math inline"><em>u</em></span> from a uniform distribution between 0 and 1. If <span class="math inline"><em>u</em> &lt; <em>R</em><sub><em>a</em><em>c</em><em>c</em><em>e</em><em>p</em><em>t</em></sub></span>, accept the proposed value of the parameter(s); otherwise reject, and retain the current value of the two parameters.</p></li>
<li><p>Repeat steps 2-5 a large number of times.</p></li>
</ol>
<p>Applying this method to the self-incompatability data, we find that again estimates of speciation and extinction differ substantially among the two character states (Figure 13.5). Since the posterior distributions for extinction do not overlap, we again infer that the character likely influences that model parameter; speciation results are again suggestive but not as conclusive as those for extinction.</p>
<div class="figure">
<img src="{{site.baseurl}}/images/figure13-5.png" alt="Figure 13.5. Bayesian BiSSe analysis of self-incompatibility. Posterior distributions for character-dependent speciation (\lambda_0 and \lambda_1) and extinction (\mu_0 and \mu_1)." />
<p class="caption">Figure 13.5. Bayesian BiSSe analysis of self-incompatibility. Posterior distributions for character-dependent speciation (<span class="math inline"><em>λ</em><sub>0</sub></span> and <span class="math inline"><em>λ</em><sub>1</sub></span>) and extinction (<span class="math inline"><em>μ</em><sub>0</sub></span> and <span class="math inline"><em>μ</em><sub>1</sub></span>).</p>
</div>
<h2 id="section-13.5potential-pitfalls-and-how-to-avoid-them">Section 13.5:Potential Pitfalls and How to Avoid Them</h2>
<p>Recently, a few papers have been published that are critical of state-dependent diversification models <span class="citation">(Rabosky and Goldberg <a href="#ref-Rabosky2015-ao">2015</a>, <span class="citation">Maddison and FitzJohn (<a href="#ref-Maddison2015-wy">2015</a>)</span>)</span>. These papers raise substantive critiques that are important to address when applying the methods described in this chapter to empirical data. In this section I will attempt to describe the critiques and their potential remedies.</p>
<p>The most serious limitation of state-dependent models as currently implemented is that they consider only a relatively small set of possible models. In particular, the approach we describe above compares two models: first, a model where birth and death rates are constant and do not depend on the state of the character; and second, a model where birth and death rates depend only on the character state <span class="citation">(Maddison et al. <a href="#ref-Maddison2007-vu">2007</a>)</span>. But there is another possibility that might be (in general) more common than either of the models we consider: birth and death rates vary, but in a way that is not dependent on the particular character we have chosen to analyze. I say that this is probably a common pattern because we know that birth and death rates vary tremendously across lineages in the tree of life <span class="citation">(Alfaro et al. <a href="#ref-Alfaro2009-hz">2009</a>)</span>, and it seems probable to me that many of our hypotheses about which characters might contribute to that variation are, at this point, stabs in the dark.</p>
<p>This issue is a normal one for statistical analyses – after all, there are always other models outside of our set of considered possibilities. However, in this case, the fact that state-dependent diversification models fail to consider the possibility outlined above causes a very particular – and peculiar – problem: if we apply the tests to empirical phylogenetic trees, even with made-up data, we almost always find statistically significant results <span class="citation">(Rabosky and Goldberg <a href="#ref-Rabosky2015-ao">2015</a>)</span>. For example, Rabosky and Goldberg <span class="citation">(<a href="#ref-Rabosky2015-ao">2015</a>)</span> found that there is very often a statistically significant “signal” that the number of letters in a species name is significantly associated with speciation rates across a range of empirical datasets. This result might seem ridiculous and puzzling, as there is no way that species name length is associated with diversification processes. However, if we return to our alternative model above, then the results make sense. Rabosky and Goldberg <span class="citation">(<a href="#ref-Rabosky2015-ao">2015</a>)</span> simulated character evolution on real phylogenetic trees, and their results do not hold when the trees are simulated along with the characters (this is also why Rabosky and Goldberg’s <span class="citation">(<a href="#ref-Rabosky2015-ao">2015</a>)</span> results do not represent “type I errors,” <em>contra</em> their paper, because the data are not simulated under the null hypothesis). On these real trees, speciation and/or extinction rates have vary across clades. Among the two models that the authors consider, both are wrong; speciation and extinction are independent of the character but not constant through time. Of the two alternatives, the state-dependent model tends to fit better because, from a statistical point of view, it is important for the model to capture some variation in birth and death rates across clades. Even a random character will pick up some of this variation, so that the alternative model tends to fit better than the null – even though, in this case, the character has nothing to do with diversification!</p>
<p>Fortunately, there are a number of ways to deal with this problem. First, one can compare the statistical support for the state-dependent model with the support that one obtains for random data. The random data could be simulated on the tree, or one could permute the tips or draw random data from a multinomial distribution <span class="citation">(Rabosky and Goldberg <a href="#ref-Rabosky2015-ao">2015</a>)</span>. One can then compare, for example, the distribution of dAICc scores obtained from these permutations to the dAICc for the original data. There are also semi-parametric methods based on permutations that have similar statistical properties <span class="citation">(Rabosky and Goldberg <a href="#ref-Rabosky2017-aw">2017</a>)</span>. Alternatively, we could explicitly consider the possibility that some unmeasured character is actually the thing that is influencing diversification rates <span class="citation">(Beaulieu and O’Meara <a href="#ref-Beaulieu2016-ww">2016</a>)</span>. This latter approach is the most elegant as we can directly add the model described in this section to our list of candidates <span class="citation">(see Beaulieu and O’Meara <a href="#ref-Beaulieu2016-ww">2016</a>)</span>.</p>
<p>A more general critique of state-dependent models of diversification was raised by Maddison and Fitzjohn <span class="citation">(Maddison and FitzJohn <a href="#ref-Maddison2015-wy">2015</a>)</span>. This paper pointed out that statistically significant results for these tests can be driven by an event on a single branch of a tree, and therefore be unreplicated. This is a good criticism that applies equally well to a range of comparative methods. Together, both of these critiques argue for a stronger set of model adequacy approaches in comparative methods.</p>
<h2 id="section-13.4-summary">Section 13.4: Summary</h2>
<p>Many evolutionary models postulate a link between species characteristics and speciation, extinction, or both. These hypotheses can be tested using state-dependent diversification models, which explicitly consider the possibility that species’ characters affect their diversification rates. State-dependent models as currently implemented have some potential problems, but there are methods to deal with these critiques. The overall ability of state-dependent models to explain broad patterns of evolutionary change remains to be determined, but represents a promising avenue for future research.</p>
<h2 id="footnotes">Footnotes</h2>
<h1 id="references" class="unnumbered">References</h1>
<div id="refs" class="references">
<div id="ref-Alfaro2009-hz">
<p>Alfaro, M. E., F. Santini, C. Brock, H. Alamillo, A. Dornburg, D. L. Rabosky, G. Carnevale, and L. J. Harmon. 2009. Nine exceptional radiations plus high turnover explain species diversity in jawed vertebrates. Proceedings of the National Academy of Sciences 106:13410–13414. National Acad Sciences.</p>
</div>
<div id="ref-Anders_Nilsson1992-af">
<p>Anders Nilsson, L. 1992. Orchid pollination biology. Trends Ecol. Evol. 7:255–259.</p>
</div>
<div id="ref-Bateman1952-iy">
<p>Bateman, A. J. 1952. Self-incompatibility systems in angiosperms. Heredity 6:285. The Genetical Society of Great Britain.</p>
</div>
<div id="ref-Beaulieu2016-ww">
<p>Beaulieu, J. M., and B. C. O’Meara. 2016. Detecting hidden diversification shifts in models of Trait-Dependent speciation and extinction. Syst. Biol. 65:583–601.</p>
</div>
<div id="ref-FitzJohn2012-tu">
<p>FitzJohn, R. G. 2012. Diversitree: Comparative phylogenetic analyses of diversification in R. Methods Ecol. Evol. 3:1084–1092.</p>
</div>
<div id="ref-FitzJohn2009-sg">
<p>FitzJohn, R. G., W. P. Maddison, and S. P. Otto. 2009. Estimating trait-dependent speciation and extinction rates from incompletely resolved phylogenies. Syst. Biol. 58:595–611. sysbio.oxfordjournals.org.</p>
</div>
<div id="ref-Goldberg2012-gs">
<p>Goldberg, E. E., and B. Igić. 2012. Tempo and mode in plant breeding system evolution. Evolution 66:3701–3709. Wiley Online Library.</p>
</div>
<div id="ref-Goldberg2010-ua">
<p>Goldberg, E. E., J. R. Kohn, R. Lande, K. A. Robertson, S. A. Smith, and B. Igić. 2010. Species selection maintains self-incompatibility. Science 330:493–495.</p>
</div>
<div id="ref-Hagey2017-hy">
<p>Hagey, T. J., J. C. Uyeda, K. E. Crandell, J. A. Cheney, K. Autumn, and L. J. Harmon. 2017. Tempo and mode of performance evolution across multiple independent origins of adhesive toe pads in lizards. Evolution 71:2344–2358.</p>
</div>
<div id="ref-Holsinger1984-ts">
<p>Holsinger, K. E., M. W. Feldman, and F. B. Christiansen. 1984. The evolution of Self-Fertilization in plants: A population genetic model. Am. Nat. 124:446–453. [University of Chicago Press, American Society of Naturalists].</p>
</div>
<div id="ref-Igic2006-zg">
<p>Igic, B., and J. R. Kohn. 2006. Bias in the studies of outcrossing rate distributions. Evolution 60:1098–1103.</p>
</div>
<div id="ref-Maddison2015-wy">
<p>Maddison, W. P., and R. G. FitzJohn. 2015. The unsolved challenge to phylogenetic correlation tests for categorical characters. Syst. Biol. 64:127–136.</p>
</div>
<div id="ref-Maddison2007-vu">
<p>Maddison, W. P., P. E. Midford, S. P. Otto, and T. Oakley. 2007. Estimating a binary character’s effect on speciation and extinction. Syst. Biol. 56:701–710. Oxford University Press.</p>
</div>
<div id="ref-Rabosky2017-aw">
<p>Rabosky, D. L., and E. E. Goldberg. 2017. FiSSE: A simple nonparametric test for the effects of a binary character on lineage diversification rates. Evolution 71:1432–1442.</p>
</div>
<div id="ref-Rabosky2015-ao">
<p>Rabosky, D. L., and E. E. Goldberg. 2015. Model inadequacy and mistaken inferences of trait-dependent speciation. Syst. Biol. 64:340–355.</p>
</div>
<div id="ref-Schopfer1999-cm">
<p>Schopfer, C. R., M. E. Nasrallah, and J. B. Nasrallah. 1999. The male determinant of self-incompatibility in brassica. Science 286:1697–1700.</p>
</div>
<div id="ref-Stebbins1950-aj">
<p>Stebbins, G. L. 1950. Variation and evolution in plants. Geoffrey Cumberlege.; London.</p>
</div>
</div>
</body>
</html>
